### PR TITLE
add retrival reward idea proof of concept

### PR DESCRIPTION
**Pull request by AlexWortega**
Here is a working proof of concept of my idea about retrival reward, its using sentence transformer to compare generated answer and ground true answer.
WB metrics
https://wandb.ai/alexwortega/trl/runs/cszh2ve8?workspace=user-alexwortega

### CODE REVIEW

**Comment by kashif** [comment]
@AlexWortega in your `sft` function I believe you are missing an `optimizer.zero_grad()`

**Comment by lvwerra** [comment]
Closing for now since there is not a lot of activity right now. Feel free to reopen :)
